tsdb/compact.go

```diff
@@ -548,10 +548,15 @@ func (c *LeveledCompactor) Write(dest string, b BlockReader, mint, maxt int64, p
 	meta.Compaction.Level = 1
 	meta.Compaction.Sources = []ulid.ULID{uid}
 
-	if parent != nil {
+	ooo := false
+	if base != nil {
 		meta.Compaction.Parents = []BlockDesc{
-			{ULID: parent.ULID, MinTime: parent.MinTime, MaxTime: parent.MaxTime},
+			{ULID: base.ULID, MinTime: base.MinTime, MaxTime: base.MaxTime},
 		}
+		ooo = base.Compaction.FromOutOfOrder()
```

Comment:
if base.Compaction.F
```suggestion
		if base.Compaction.FromOutOfOrder() {
		  meta.Compaction.SetOutOfOrder()
		}
```